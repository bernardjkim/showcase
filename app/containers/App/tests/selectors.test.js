import { fromJS } from 'immutable';

import {
  selectGlobalDomain,
  makeSelectLocation,
  makeSelectToken,
  makeSelectUser,
  makeSelectError,
  makeSelectLoading,
} from 'containers/App/selectors';

describe('makeSelectLocation', () => {
  const locationStateSelector = makeSelectLocation();
  it('should select the location', () => {
    const mockedState = fromJS({
      router: { location: { pathname: '/foo' } },
    });
    expect(locationStateSelector(mockedState)).toEqual(
      mockedState.getIn(['router', 'location']).toJS(),
    );
  });
});

const initialState = fromJS({
  loading: false,
  error: false,
  token: false,
  user: false,
});

describe('selectGlobalDomain', () => {
  it('should select the global state', () => {
    expect(selectGlobalDomain(initialState)).toMatchSnapshot();
  });
});

describe('makeSelectLoading', () => {
  const loadingSelector = makeSelectLoading();
  it('should select the loading state', () => {
    expect(loadingSelector(initialState)).toMatchSnapshot();
  });
});

describe('makeSelectError', () => {
  const errorSelector = makeSelectError();
  it('should select the error state', () => {
    expect(errorSelector(initialState)).toMatchSnapshot();
  });
});

describe('makeSelectToken', () => {
  const tokenSelector = makeSelectToken();
  it('should select the token state', () => {
    expect(tokenSelector(initialState)).toMatchSnapshot();
  });
});

describe('makeSelectUser', () => {
  const userSelector = makeSelectUser();
  it('should select the user state', () => {
    expect(userSelector(initialState)).toMatchSnapshot();
  });
});
