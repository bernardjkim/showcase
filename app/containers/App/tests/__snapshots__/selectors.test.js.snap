// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`makeSelectError should select the error state 1`] = `false`;

exports[`makeSelectLoading should select the loading state 1`] = `false`;

exports[`makeSelectToken should select the token state 1`] = `false`;

exports[`makeSelectUser should select the user state 1`] = `false`;

exports[`selectGlobalDomain should select the global state 1`] = `
Immutable.Map {
  "loading": false,
  "error": false,
  "token": false,
  "user": false,
}
`;
